compute/cker/include/cker/train/operation/ReLU6.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NNFW_CKER_TRAIN_OPERATION_RELU6_H__
+#define __NNFW_CKER_TRAIN_OPERATION_RELU6_H__
+
+#include "cker/Shape.h"
+#include "cker/eigen/Utils.h"
+#include <Eigen/Core>
+
+namespace nnfw
+{
+namespace cker
+{
+namespace train
+{
+
+inline void ReLU6Grad(const Shape &output_shape, const float *output_data,
+                      const Shape &incoming_shape, const float *incoming_data,
+                      const Shape &grad_shape, float *grad_data)
+{
+  const auto output_map = MapAsVector(output_data, output_shape);
+  const auto incoming_map = MapAsVector(incoming_data, incoming_shape);
+  auto grad_map = MapAsVector(grad_data, grad_shape);
```

Comment:
Q. Is this ref var? or just local var?

Reply:
> ref var? or just local var? 

Umm.. I failed to catch your question. 
Does ref var mean reference( = original data is not in function stack) and local var mean the one in which data resides in the function stack?  

-------------------------------
I don't know if it is the right answer to your question. 
Anyway I put my understanding about the variable here : 

The variable `grad_map` holds `Eigen::Map` object. 
`Eigen::Map` is an object to provide a matrix(sometimes vector) view to the existing array. 

* document about eigen map : https://eigen.tuxfamily.org/dox/classEigen_1_1Map.html#:~:text=PlainObjectType%2C%20MapOptions%2C%20StrideType%20%3E-,A%20matrix%20or%20vector%20expression%20mapping%20an%20existing%20array%20of%20data.,-Template%20Parameters

Reply:
Sorry for confusing. Just curious.

My point was what `auto` of `auto grad_map` is. `Eigen::Map grad_map` or `Eigen::Map &grad_map` or `&&grad_map` could. I'm curious whether unnecessary copying happens.

Reply:
AFAIK, Copy doesn't happen. This function just changes the view of the same memory data.

Reply:
I also did some search about this.

After your comment, 
I worried that since `MapAsObject` returns an object, `auto grad_map= ..` might call the copy constructor.
But as I understood, this doesn't call a redundant copy constructor, because of compiler optimization called - RVO (Return value optimization).

Reply:
Great. That is exactly what I want to know. Thanks!